### PR TITLE
Fix error when saving needle JSON with non-ASCII characters

### PR DESCRIPTION
**Pull request by Martchus**
Otherwise saving needles fails with the error message `syntax error: malformed UTF-8 character in JSON string`.

(Noticed when testing https://github.com/os-autoinst/openQA/pull/2070.)

### CODE REVIEW

File: lib/OpenQA/Task/Needle/Save.pm [hunk 1]
```diff
@@ -69,7 +70,7 @@ sub _save_needle {
     my $schema       = $app->schema;
     my $openqa_job   = $schema->resultset('Jobs')->find($args->{job_id});
     my $user         = $schema->resultset('Users')->find($args->{user_id});
-    my $needle_json  = $args->{needle_json};
+    my $needle_json  = encode_utf8($args->{needle_json});
```
Comment (kalikiana) [inline lib/OpenQA/Task/Needle/Save.pm]: Can we add a JSON with UTF-8 characters for this to `t/16-utils-runcmd.t`?

**Comment by Amrysliu** [comment]
LGTM

File: lib/OpenQA/Task/Needle/Save.pm (hunk 1 above)
Comment (Martchus) [reply lib/OpenQA/Task/Needle/Save.pm]: I added tests. But I added them in `t/ui/12-needle-editor.t` so the fronted parts are tested as well.

Comment (kalikiana) [reply lib/OpenQA/Task/Needle/Save.pm]: Good call!

**Pull request merged**